### DISCUSSION TITLE
How to read nodeset from the file?

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#20939] [Q&A Meshing] [answered]
Hello MOOSE Experts

I want to know if there is any way to read the NODESETS from the file different from the mesh input file.

Lets say I have 4 files, for which I want to generate 4 NODESETS means each nodeset for each file.

Any suggestions?

Thanks in advance.

Best
Avtar

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can read 4 mesh files and then stitch them or just combine them if they are separate.
You cant read most of the mesh from one file and the nodesets from another currently. 

But you could also implement a [mesh generator](https://mooseframework.inl.gov/syntax/Mesh/index.html) that does that, it's not very hard. 

Guillaume

**Reply by avtarsinghh1991** [reply]
Thanks. I tried to generate the ExtraNodesetGenerator by giving the node numbers but its nor working properly as nodes and element renumbering occurs. I tried allow_renumbering = false but still its not working.

Any suggestions?

**Reply by GiudGiud** [reply]
no that should work
what kind of simulation are you doing? distributed? split mesh? renumbering only happens on selective occasions

**Reply by GiudGiud** [reply]
how did you get the node numbers? From the exodus file?

**Reply by avtarsinghh1991** [reply]
Thanks. Its a structured mesh contains multiple grains. I wrote the .msh file using matlab subroutine. And from the matlab subroutine itself I got to know the node numbers for nodesets. 

Please find the attached mesh file (.msh) and the moose input file of mesh (.i)
[File.zip](https://github.com/idaholab/moose/files/8621850/File.zip)
.

**Reply by GiudGiud** [reply]
then why dont you include the nodesets in the .msh file?

**Reply by avtarsinghh1991** [reply]
Thanks Guillaume.

Yes, I can do that. I only thing is that I am still figuring out the syntax to add the nodesets in .msh file. However, thanks for your suggestions.

Best
Avtar

**Reply by GiudGiud** [reply]
Sounds good. If you find it easier to add a side set in gmsh you can have moose convert a side set to a node set. But I suspect node set is easier in gmsh

**Reply by avtarsinghh1991** [reply]
@GiudGiud 

Can I provide two files i.e., .msh and .geo to MOOSE for mesh data and nodeset data?

**Reply by GiudGiud** [reply]
not for the same parts of the domain currently.
